### PR TITLE
feat: add version deprecation error-code identification logic

### PR DESCRIPTION
**Pull request by sanpj2292**
# Description

The idea is to identify version deprecation error category from error message & store it for reporting purposes

## Linear Ticket

< Replace with Linear Link >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: enterprise/reporting/error_extractor.go [hunk 1]
```diff
@@ -31,21 +32,52 @@ var (
...
-	log              logger.Logger
-	ErrorMessageKeys []string // the keys where in we may have error message
+	log                              logger.Logger
+	ErrorMessageKeys                 []string // the keys where in we may have error message
+	versionDeprecationThresholdScore misc.ValueLoader[int]
 }
 
 func NewErrorDetailExtractor(log logger.Logger) *ExtractorHandle {
 	errMsgKeys := config.GetStringSlice("Reporting.ErrorDetail.ErrorMessageKeys", []string{})
+	versionDepThreshold := config.GetReloadableIntVar(1, 1, "Reporting.ErrorDetail.versionDeprecationThresholdScore")
```
Comment (koladilip) [inline enterprise/reporting/error_extractor.go:73]: should we keep min score to be 2 as version keep might come in different context also

Comment (sanpj2292) [reply enterprise/reporting/error_extractor.go:73]: This is a reloadable configuration variable. We can put the desired value in operator

**Review by koladilip — APPROVED** [review]

File: enterprise/reporting/error_extractor.go [hunk 2]
```diff
@@ -255,3 +287,22 @@ func (ext *ExtractorHandle) CleanUpErrorMessage(errMsg string) string {
 
 	return regexdMsg
 }
+
+func (ext *ExtractorHandle) GetErrorCode(errorMessage string) string {
+	// version deprecation logic
+	var score int
+	var errorCode string
+
+	// Check if the value is a string before converting to lowercase
```
Comment (chandumlg) [inline enterprise/reporting/error_extractor.go:296]: Remove this comment.

File: enterprise/reporting/error_extractor.go [hunk 3]
```diff
@@ -31,21 +32,52 @@ var (
 	whitespacesRegex = regexp.MustCompile("[ \t\n\r]*") // used in checking if string is a valid json to remove extra-spaces
 
 	defaultErrorMessageKeys = []string{"message", "description", "detail", "title", errorKey, "error_message"}
+	deprecationKeywords     = map[string]int{
```
Comment (chandumlg) [inline enterprise/reporting/error_extractor.go:35]: Can we ensure lowercase of keys somehow (to make sure we don't have casing issues for the keys added in the future)?

**Review by chandumlg — APPROVED** [review]
Approving with minor comments.

File: enterprise/reporting/error_extractor.go (hunk 3 above)
Comment (sanpj2292) [reply enterprise/reporting/error_extractor.go:35]: Are you asking to deliberately lowercase even though this map now contains lowercased keys ?

Comment (sanpj2292) [reply enterprise/reporting/error_extractor.go:35]: Done

**Pull request merged**